waku/protocol/v2/waku_store.nim

```diff
@@ -124,6 +124,10 @@ method init*(T: type WakuStore): T =
   result = ws
 
 proc filter*(proto: WakuStore): Filter =
+  ## The filter function returns the pubsub filter for the node.
```

Comment:
Filter function... pubsub filter (?)...

IMO this is confusing, especially since we already have a clear concept of a filter in the spec. Surely there's a better way of encapsulating this idea?

Reply:
I still don't understand why we aren't using the subscribe/subscribeTopic handler in WakuSub/WakuRelay here. Isn't that where this logic belongs?

cc @kdeme

Reply:
I'm not sure what you are saying, this filter has nothing to do with the spec. Its an implementation detail. Designed to share information between loosely coupled components of the node.